testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveStorageFormats.java

```diff
@@ -542,6 +549,109 @@ public void testOrcTableCreatedInTrino()
+                                ? Timestamp.valueOf(LocalDateTime.of(2015, 5, 10, 6, 30, 35, 123_000_000))
+                                : Timestamp.valueOf(LocalDateTime.of(2015, 5, 10, 12, 15, 35, 123_000_000))));
+            }
+        }
+        else {
+            columnDataList.add(
+                    new StorageFormatHiveCompatibilityColumnData(
+                            "c_timestamp",
+                            "timestamp",
+                            "TIMESTAMP '2015-05-10 12:15:35.123'",
+                            Timestamp.valueOf(LocalDateTime.of(2015, 5, 10, 12, 15, 35, 123_000_000))));
+        }
+        columnDataList.add(new StorageFormatHiveCompatibilityColumnData("c_array", "array(integer)", "ARRAY[1, 2, 3]", "[1,2,3]"));
+        columnDataList.add(new StorageFormatHiveCompatibilityColumnData("c_map", "map(varchar, varchar)", "MAP(ARRAY['foo'], ARRAY['bar'])", "{\"foo\":\"bar\"}"));
+        columnDataList.add(new StorageFormatHiveCompatibilityColumnData("c_row", "row(f1 integer, f2 varchar)", "ROW(42, 'Trino')", "{f1:42,f2:\"Trino\"}"));
+
+        onTrino().executeQuery(
+                format("CREATE TABLE %s (" +
+                                "%s" +
+                                ") " +
+                                "WITH (%s)",
+                        tableName,
+                        columnDataList.stream()
+                                .map(data -> format("%s %s", data.columnName, data.trinoColumnType))
+                                .collect(Collectors.joining(", \n")),
+                        storageFormat.getStoragePropertiesAsSql()));
+
+        onTrino().executeQuery(
+                format(
+                        "INSERT INTO %s VALUES (%s)",
+                        tableName,
+                        columnDataList.stream()
+                                .map(data -> data.trinoInsertValue)
+                                .collect(Collectors.joining(", \n"))));
+
+        // array, map and struct fields are interpreted as strings in the hive jdbc driver and need therefore special handling
+        Function<StorageFormatHiveCompatibilityColumnData, Boolean> columnsInterpretedCorrectlyByHiveJdbcDriverPredicate = data ->
+                !ImmutableList.of("c_array", "c_map", "c_row").contains(data.columnName);
+        QueryResult queryResult = onHive().executeQuery(
+                format("SELECT %s FROM %s",
+                        columnDataList.stream()
+                                .filter(columnsInterpretedCorrectlyByHiveJdbcDriverPredicate::apply)
+                                .map(data -> data.columnName)
+                                .collect(Collectors.joining(", \n", "", "\n")),
+                        tableName));
+        assertThat(queryResult).containsOnly(new Row(
+                columnDataList.stream()
+                        .filter(columnsInterpretedCorrectlyByHiveJdbcDriverPredicate::apply)
```

Comment:
what happens if  you remove this filter?

Reply:
In the case we'd delete the predicate, the assertion will fail 
```
        assertThat(queryResult).containsOnly(new Row(
                columnDataList.stream()
                        .filter(columnsInterpretedCorrectlyByHiveJdbcDriverPredicate::apply)
                        .map(data -> data.hiveJdbcExpectedValue)
                        .collect(toImmutableList())));
```


The reason for the fail is because
```
io.trino.tempto.assertions.QueryResultValueComparator#arrayEqual
```
expects to deal with an `Array`, however the hive jdbc driver reads "String" for complex types (`map`, `struct`, aarray`).

https://stackoverflow.com/a/43191428

Reply:
I know that Hive JDBC returns complex types as Strings
what is the `JDBCType` that `QueryResultValueComparator` sees? for these? 

also, since we have one row anyway we can switch over to AssertJ here (with explanatory comment).
we should make the code more complex just to use tempto's assertions.

Reply:
Debugging into `io.trino.tempto.assertions.QueryResultValueComparator#arrayEqual` shows that the compared values are of  `String` type.

```
if (actual instanceof Array && expected instanceof List) {
...
} else {
    return false;
}
```

Reply:
> also, since we have one row anyway we can switch over to AssertJ here (with explanatory comment).
> we should make the code more complex just to use tempto's assertions.

I am sorry I don't quite follow. Can you provide me a hint to what you are actually suggesting?